packages/flutter/lib/src/material/time_picker.dart

```diff
@@ -1639,7 +1660,7 @@ class _HourMinuteTextFieldState extends State<_HourMinuteTextField> {
   @override
   void didChangeDependencies() {
     super.didChangeDependencies();
-    controller ??= TextEditingController(text: _formattedValue);
+    controller.value.text = _formattedValue;
```

Comment:
@darrenaustin @rami-a I know that the `controller` was originally defined this way because it needs access to the `BuildContext` to get localized text values here. The state restoration framework needs `controller` to be defined after `initState` but before `didChangeDependencies` and I was wondering if this change would be okay. Otherwise, I'm open to suggestions on how to improve this